nixos/modules/services/home-automation/openthread-border-router.nix

```diff
@@ -0,0 +1,266 @@
+                  "--thread-ifname ${cfg.interfaceName}"
+                  "--debug-level ${toString logLevel}"
+                ]
+                (lib.optional (cfg.rest.listenPort != 0) "--rest-listen-port ${toString cfg.rest.listenPort}")
+                (lib.optional (cfg.rest.listenAddress != "") "--rest-listen-address ${cfg.rest.listenAddress}")
+                [ cfg.radio.url ]
+                cfg.radio.extraDevices
+              ]
+            )
+          );
+          ExecStopPost = "${lib.getExe' cfg.package "otbr-firewall"} stop";
+          KillMode = "mixed";
+          Restart = "on-failure";
+          RestartSec = 5;
+          RestartPreventExitStatus = "SIGKILL";
+
+          # Hardening options (not present in upstream service definitions)
+          StateDirectory = "thread";
+          ProtectSystem = "strict";
+          ProtectHome = true;
+          PrivateTmp = true;
+          ProtectKernelLogs = true;
+          ProtectControlGroups = true;
+          NoNewPrivileges = true;
+          LockPersonality = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          SystemCallArchitectures = "native";
+
+          CapabilityBoundingSet = [
+            "CAP_NET_ADMIN"
+            "CAP_NET_RAW"
+          ];
+        };
+        path = [
+          pkgs.ipset
+          pkgs.iptables
+        ];
+      };
+
+      # Sync with: src/web/otbr-web.service.in
+      otbr-web = lib.mkIf cfg.web.enable {
+        description = "OpenThread Border Router Web Interface";
+        after = [ "otbr-agent.service" ];
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          ExecStart = (
+            lib.concatStringsSep " " (
+              lib.concatLists [
+                [
+                  (lib.getExe' cfg.package "otbr-web")
+                  "-I"
+                  "${cfg.interfaceName}"
+                  "-d"
+                  "${toString logLevel}"
+                ]
+                (lib.optional (cfg.web.listenAddress != "") "-a ${cfg.web.listenAddress}")
+                (lib.optional (cfg.web.listenPort != 0) "-p ${toString cfg.web.listenPort}")
+              ]
+            )
+          );
+
+          # Hardening options (not present in upstream service definitions)
+          DynamicUser = true;
+          ProtectSystem = "strict";
+          ProtectHome = true;
+          PrivateTmp = true;
+          PrivateDevices = true;
+          ProtectKernelLogs = true;
+          ProtectKernelTunables = true;
+          ProtectKernelModules = true;
+          ProtectControlGroups = true;
+          NoNewPrivileges = true;
+          LockPersonality = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          RestrictNamespaces = true;
+          SystemCallArchitectures = "native";
+          RestrictAddressFamilies = [
+            "AF_INET"
+            "AF_INET6"
+          ];
+          CapabilityBoundingSet = "";
```

Comment:
There is a problem with these restrictions.

According to https://github.com/openthread/ot-br-posix/blob/main/src/web/web-service/ot_client.cpp#L94, otbr-web connects to daemon over AF_UNIX, to socket in `/run/openthread-wpan0.sock`

Reply:
Thanks for testing! I've tested and updated the otbr-web service.